sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberTree.kt

```diff
@@ -0,0 +1,102 @@
+package io.sentry.android.timber
+
+import android.util.Log
+import io.sentry.core.Breadcrumb
+import io.sentry.core.IHub
+import io.sentry.core.SentryEvent
+import io.sentry.core.SentryLevel
+import io.sentry.core.protocol.Message
+import timber.log.Timber
+
+/**
+ * Sentry Timber tree which is responsible to capture events via Timber
+ */
+class SentryTimberTree(
+    private val hub: IHub,
+    private val minEventLevel: SentryLevel,
+    private val minBreadcrumbLevel: SentryLevel
+) : Timber.Tree() {
+
+    /**
+     * do not log if it's lower than min. required level.
+     */
+    private fun isLoggable(level: SentryLevel, minLevel: SentryLevel): Boolean = level.ordinal >= minLevel.ordinal
+
+    /**
+     * Compares the minLevel given by the user and the level given by the event
+     * to capture an event or not.
+     */
+    override fun isLoggable(tag: String?, priority: Int): Boolean {
+        val level = getSentryLevel(priority)
+
+        // checks only the event level
+        return isLoggable(level, minEventLevel)
+    }
+
+    /**
+     * Captures a Sentry Event if the min. level is equal or higher than the min. required level.
+     */
+    override fun log(priority: Int, tag: String?, message: String, throwable: Throwable?) {
+        val level = getSentryLevel(priority)
+
+        captureEvent(level, tag, message, throwable)
+        addBreadcrumbIfEventHasThrowable(level, throwable, message)
+    }
+
+    /**
+     * Captures an event with the given attributes
+     */
+    private fun captureEvent(level: SentryLevel, tag: String?, message: String, throwable: Throwable?) {
+        val sentryEvent = SentryEvent()
+        sentryEvent.level = level
+
+        // if there's no throwable, should it be a breadcrumb then?
+        throwable?.let {
+            sentryEvent.setThrowable(it)
+        }
+        sentryEvent.message = Message().apply {
+            formatted = message
+        }
+
+        tag?.let {
+            sentryEvent.setTag("TimberTag", it)
+        }
+        // maybe this should add itself to the integration package
+        // we need to define a merging mechanism then
+        sentryEvent.setTag("origin", "SentryTimberIntegration")
+
+        hub.captureEvent(sentryEvent)
+    }
+
+    /**
+     * Adds a breadcrumb if the event has an exception.
+     */
+    private fun addBreadcrumbIfEventHasThrowable(level: SentryLevel, throwable: Throwable?, message: String) {
```

Comment:
H: should this just be add Breadcrumb? We should add it regardless of having throwable.

Reply:
well, this breadcrumb is `exception` category and `error` as a type, so it'd make sense to be only if there's an error.
if we log everything as a breadcrumb, I can't choose a category and type as they won't match most of the time, so I'm not really sure if I'd like to log literally everything as a breadcrumb.
If doing that, I'd keep type and category just not set at all, wdyt?

Reply:
Other logging integrations log everything, without any specific category that I can recall. It just goes blank (it's not required). The `level` is more important.

It's very useful to see the last 100 log entries (i.e Info or higher) right before an event.